tests/integration/test_metrics.py

```diff
@@ -39,6 +40,21 @@ def metrics_by_name(metrics_consumer, count, timeout=None):
     return metrics
 
 
+def metrics_by_name_group_by_project(metrics_consumer, count, timeout=None):
```

Comment:
This helper function does not seem to be using `count` at all. I feel like this function could call `metrics_by_name` internally, or vice versa?

Reply:
count is huge antipattern IMO, I'll remove it from args :)

Reply:
let me explain, this helper has small problem:
```
def metrics_by_name(metrics_consumer, count, timeout=None):
    metrics = {}

    for _ in range(count):
        metric = metrics_consumer.get_metric(timeout)
        metrics[metric["name"]] = metric

    metrics_consumer.assert_empty()
    return metrics
```
if you have 2 projects and 2 metrics with the same name you will see only the last one, so that's why I decided to create a new helper function.

Reply:
And the second problem, you need manually to set the count - I prefer to get all metrics - since we have this assert_empty and then return data.